src/main/java/edu/carroll/dogtag/web/controller/TrainingController.java

```diff
@@ -0,0 +1,103 @@
+package edu.carroll.dogtag.web.controller;
+
+import edu.carroll.dogtag.jpa.model.Training;
+import edu.carroll.dogtag.service.LoginService;
+import edu.carroll.dogtag.service.TrainingService;
+import edu.carroll.dogtag.service.UserProfileService;
+import edu.carroll.dogtag.web.form.TrainingForm;
+import jakarta.servlet.http.HttpSession;
+import jakarta.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.List;
+
+/**
+ * The TrainingController handles the information that is entered in the html pages to the LoginForm
+ * and calls methods to check if there are training logs in the database for the user that is logged in
+ * and then adds the list to the model to be passed back to the html page to be displayed.  When a new log
+ * is saved and the page is reloaded the training logs are fetched again in a list format, and it is then updated
+ * on the html page.
+ */
+@Controller
+public class TrainingController {
+    private static final Logger log = LoggerFactory.getLogger(TrainingController.class);
+    private final TrainingService trainingService;
+    private final LoginService loginService;
+    private final UserProfileService userProfileService;
+
+    public TrainingController(TrainingService trainingService, LoginService loginService, UserProfileService userProfileService) {
+        this.trainingService = trainingService;
+        this.loginService = loginService;
+        this.userProfileService = userProfileService;
+    }
+
+    /**
+     * @param session It allows the server to store and retrieve
+     *                user-specific data between requests.
+     * @return This class merely holds both to make it possible for a controller
+     * to return both model and view in a single return value. ModelAndView is a value object designed
+     * to hold model and view making it possible for a handler to return both model
+     * and view in a single return value. In this case primarily a list of trainings for
+     * the intended user that is being passed to fetchUserTraining(user).  This allows the
+     * template to immediately display the new information that has been entered.
+     */
+    @GetMapping("/traininglog")
+    public String trainingForm(HttpSession session, Model model) {
+        final String user = (String) session.getAttribute("user");
+        if (user == null || user.isBlank()) {
+            log.error("Login user {} does not have session", user);
```

Comment:
I don't think is an error, as an error indicates a developer issue, and someone trying to get to the traininglog page and not being logged in is probably NOT a developer error.